crates/provider-wit-bindgen/src/lib.rs

```diff
@@ -41,187 +55,926 @@ const EXPORTS_MODULE_NAME: &str = "exports";
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+enum WitFunctionLatticeTranslationStrategy {
+    /// Automatically determine how to bundle a function and it's arguments to be sent over the
+    ///
+    /// Generally, this means detecting whether the function has a single argument or multiple
+    /// and choosing whether to send that single argument or bundle up all arguments into a generated Invocation struct.
+    #[default]
+    Auto,
+
+    /// First argument usage assumes that every function that should be exported to the lattice
+    /// has *one* argument, which is the object that should be serialized and set out onto the lattice.
+    ///
+    /// For example, the following WIT:
+    ///
+    /// ```ignore
+    /// interface example {
+    ///   f: func(input: string);
+    ///
+    ///   record g-request {
+    ///     first: string;
+    ///     second: u32;
+    ///     third: bool;
+    ///   }
+    ///
+    ///   g: func(input: g-request);
+    ///
+    ///   h: func(first: bool, second: string);
+    /// }
+    /// ```
+    ///
+    /// Under this setting, bindgen will produce an error on function `h`, as it contains multiple arguments
+    FirstArgument,
+
+    /// Argument bundling assumes that every function that should be exported to the lattice
+    /// has *one or more* arguments, which are bundled into an object that should be serialized and set out onto the lattice.
+    ///
+    /// For example, the following WIT:
+    ///
+    /// ```ignore
+    /// package examples:arg-bundle;
+    ///
+    /// interface example {
+    ///   f: func(input: string);
+    ///
+    ///   record g-request {
+    ///     first: string;
+    ///     second: u32;
+    ///     third: bool;
+    ///   }
+    ///
+    ///   g: func(input: g-request);
+    ///
+    ///   h: func(first: bool, second: string);
+    /// }
+    /// ```
+    ///
+    /// Under this setting, bindgen will not produce an error on any function, but will be slightly
+    /// inefficient as it will wrap `f` in a generated `ExamplesArgBundleFInvocation` struct (containing one member, `input`).
+    BundleArguments,
+}
+
+impl WitFunctionLatticeTranslationStrategy {
+    /// Translate a wit-bindgen generated trait function for use across the lattice
+    fn translate_import_fn_for_lattice(
+        &self,
+        wit_iface_name: WitInterfaceName,
+        trait_method: &TraitItemMethod,
+        struct_lookup: &StructLookup,
+        type_lookup: &TypeLookup,
+        bindgen_cfg: &ProviderBindgenConfig,
+    ) -> anyhow::Result<(WitInterfaceName, LatticeMethod)> {
+        // Calculate the lattice method name
+        let lattice_method_name = LitStr::new(
+            match (
+                bindgen_cfg.contract.as_str(),
+                &bindgen_cfg.wit_ns,
+                &bindgen_cfg.wit_pkg,
+            ) {
+                // To facilitate backwards compatibility with existing wasmcloud:keyvalue 
+                ("wasmcloud:keyvalue", _, _) => {
```

Comment:
Does this mean we'll need to add each legacy wasmCloud interface here as we go?

Reply:
yep it would -- the contract names would have had to be added, but also note that for now I'm going with the conversion of the interface name!